share/completions/go.fish

```diff
@@ -36,18 +45,28 @@ complete -c go -n __fish_use_subcommand -a build -d 'compile packages and depend
 
 # clean
 complete -c go -n __fish_use_subcommand -a clean -d 'remove object files'
-complete -c go -n '__fish_seen_subcommand_from clean' -s i -d "remove the corresponding installed archive or binary (what 'go install' would create)"
+complete -c go -n '__fish_seen_subcommand_from clean' -s i -d "remove the corresponding installed archive or binary"
 complete -c go -n '__fish_seen_subcommand_from clean' -s n -d "print the remove commands it would execute, but not run them"
-complete -c go -n '__fish_seen_subcommand_from clean' -s r -d "clean to be applied recursively to all the dependencies of the packages named by the import paths"
+complete -c go -n '__fish_seen_subcommand_from clean' -s r -d "clean to be applied recursively to all the dependencies"
 complete -c go -n '__fish_seen_subcommand_from clean' -s x -d "clean to print remove commands as it executes them"
+complete -c go -n '__fish_seen_subcommand_from clean' -o cache -d 'clean to remove the entire go build cache'
+complete -c go -n '__fish_seen_subcommand_from clean' -o testcache -d 'clean to expire all test results in the go build cache'
+complete -c go -n '__fish_seen_subcommand_from clean' -o modcache -d 'clean to remove the entire module download cache'
 
 # doc
 complete -c go -n __fish_use_subcommand -a doc -d 'run godoc on package sources'
-complete -c go -n '__fish_seen_subcommand_from doc' -s n -d "prints commands that would be executed"
-complete -c go -n '__fish_seen_subcommand_from doc' -s x -d "prints commands as they are executed"
+complete -c go -n '__fish_seen_subcommand_from doc' -o all -d "Show all the documentation for the package"
+complete -c go -n '__fish_seen_subcommand_from doc' -s c -d 'Respect case when matching symbols'
+complete -c go -n '__fish_seen_subcommand_from doc' -o cmd -d 'Treat a command (package main) like a regular package'
+complete -c go -n '__fish_seen_subcommand_from doc' -o short -d 'One-line representation for each symbol'
+complete -c go -n '__fish_seen_subcommand_from doc' -o src -d 'Show the full source code for the symbol'
+complete -c go -n '__fish_seen_subcommand_from doc' -s u -d 'Show documentation for unexported as well as exported'
 
 # env
 complete -c go -n __fish_use_subcommand -a env -d 'print Go environment information'
+complete -c go -n '__fish_seen_subcommand_from env' -o json -d 'prints the environment in JSON format'
+complete -c go -n '__fish_seen_subcommand_from env' -s u -r -d 'unsets NAME to default'
```

Comment:
"unsets NAME" means NAME isn't set afterwards.

Does it set NAME to the default, or leave NAME not set at all?

Reply:
unset NAME, and NAME will follow default GOENV